test/phoenix_html/form_test.exs

```diff
@@ -261,6 +287,18 @@ defmodule Phoenix.HTML.FormTest do
 
       assert form =~ ~s(<span class="errors">Field error message!</span>)
     end
+
+    test "with id prefix the form id in the input id" do
+      form =
+        safe_to_string(
+          form_for(:search, "/", [params: search_params(), id: "form_id"], fn f ->
+            text_input(f, :key)
+          end)
+        )
+
+      assert form =~
+               ~s(<input id="form_id_search_key" name="search[key]" type="text" value="value">)
```

Comment:
As we can see here, the form id is prefixed when using `form_for/4`. Just added this test to ensure that 🙌 

https://github.com/phoenixframework/phoenix_html/blob/700f0e096c06d19c75ccaf4a0ad017bd605dfea6/lib/phoenix_html/form_data.ex#L96